### PR TITLE
WIP RSS Feed datasource

### PR DESCRIPTION
**Pull request by dprokop**
This moves feed fetching logic from NewsPanel to a RSS Feed datasource

@ryantxu not sure this makes much sense to you as @torkelo mentioned you wanted the RSS ds to be a backend plugin. But I'm gonna leave this WIP here anyways ;)

### CODE REVIEW

File: public/app/plugins/datasource/rss-feed/types.ts [hunk 1]
```diff
@@ -1,13 +1,13 @@
-export const DEFAULT_FEED_URL = 'https://grafana.com/blog/news.xml';
+import { DataQuery, DataSourceJsonData } from '@grafana/data';
 
-export interface NewsOptions {
+export interface RSSFeedQuery extends DataQuery {}
+
+export interface RSSFeedOptions extends DataSourceJsonData {
+  // Saved in the datasource and download with bootData
```
Comment (dprokop) [inline public/app/plugins/datasource/rss-feed/types.ts]:
```suggestion
```

**Comment by ryantxu** [comment]
Yes -- I have two concerns... and paused development for a long time based on them, finally just put the fetch directly in the news panel until we can find a better solution :)

1. We want the new panel to work 'Out of the box' -- so the datasource will have to be setup and configured out of the box

2. RSS/Atom can be a super powerful, but kind of complicated API.  The frontend tools to read it are pretty finicky and there are many good/supported ways to do that on the backend.  And we would like alerting :)

**Comment by dprokop** [comment]
> And we would like alerting :)

Yeh, this is interesting :) So I assume other use cases for this datasource that just reading news list?

**Comment by ryantxu** [comment]
I come from another era where XML was the default language for most APIs... at some point things settled on an RSS like standard https://en.wikipedia.org/wiki/OpenSearch

The result is that many old APIs -- in particular spatial datasources have XML/Atom/RSS style interface

**Comment by dprokop** [comment]
Ok, thanks for clarifying @ryantxu. The I'm gonna close this PR for now :)

**Pull request closed without merging**